platform/BufferedSerial.h

```diff
@@ -85,7 +95,7 @@ class BufferedSerial : private SerialBase, public FileHandle {
     void tx_irq(void);
     void rx_irq(void);
 
-
+    void DCD_IRQ(void);
```

Comment:
IMO there is little or no sense in having a serial interface with no buffering. This _BufferedSerial_ should actually have been a _Serial_ really. Unfortunately as it stands, we have a nightmarish _Serial_ implementation. However coming back to the point, Kevin explained the stuff above, I will summarise:
 
-> BufferedSerial (I will change this ghastly name), is a FileHandle
-> FileHandle(s) need to be polled for events 
-> One of those events is HUP (hangup)
-> If and only if there is DCD line wired up (and I can't stress that more) and you gave a valid pin, only then a DCD_IRQ is set.